framework/src/main/java/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -439,24 +439,21 @@ private boolean isIntRangeOrIntegralUnknownVal(Node node, TransferInput<CFValue,
   }
 
   /**
-   * Create a new transfer result based on the original result and the new annotation.
+   * Create a new boolean transfer result based on the original result and the new annotation.
    *
-   * @param result the original result
-   * @param resultAnno the new annotation
-   * @return the new transfer result
+   * @param thenStore the then store for the result
+   * @param elseStore the else store for the result
+   * @param booleanValues the possible values that the result might evaluate to
+   * @param underlyingType the (boolean) result type
+   * @return a transfer result like {@code result}, but permitting only the given boolean values
    */
-  private TransferResult<CFValue, CFStore> createNewResult(
-      TransferResult<CFValue, CFStore> result, AnnotationMirror resultAnno) {
-    CFValue newResultValue =
-        analysis.createSingleAnnotationValue(
-            resultAnno, result.getResultValue().getUnderlyingType());
-    return new RegularTransferResult<>(newResultValue, result.getRegularStore());
-  }
-
-  /** Create a boolean transfer result. */
+  // TODO: What is the point of `underlyingType`?  Won't it always be `boolean`?
```

Comment:
There's no easy way to get a type mirror for a specific type, so it's best to just pass it if the caller already has it.